docs/reference/components/sources/dnstap.cue

```diff
@@ -0,0 +1,1325 @@
+package metadata
+
+components: sources: dnstap: {
+	title: "Dnstap"
+
+	classes: {
+		commonly_used: false
+		delivery:      "best_effort"
+		deployment_roles: ["daemon"]
+		development:   "stable"
+		egress_method: "stream"
+		stateful:      false
+	}
+
+	features: {
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.bind_dnstap
+				interface: socket: {
+					api: {
+						title: "dnstap"
+						url:   urls.dnstap
+					}
+					direction: "incoming"
+					port:      0
+					protocols: ["unix"]
+					socket: "/run/bind/dnstap.sock"
+					ssl:    "disabled"
+				}
+			}
+			tls: enabled: false
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  false
+			"armv7-unknown-linux-musleabihf": false
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          false
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration: {
+		max_frame_length: {
+			common:      false
+			description: "Max dnstap frame length that the dns source can handle."
+			required:    false
+			type: uint: {
+				default: 102400
+				unit:    "bytes"
+			}
+		}
+		socket_path: {
+			description: """
+				Absolute path of server socket file to which BIND is configured
```

Comment:
Good point, Jesse 👍 . Will do.

Reply:
> Another question @j-bai : we are curious to hear more about the motivation to add the multithreading controls here. Did you find it couldn't keep up with your test system? Do you have an idea of how many messages / second it would cap out at?

Yes, the `multithreaded` mode was introduced in to solve the CPU underload problem we noticed in the early development stage of the dnstap source component. According to our performance test (did several months ago with vector build of version `0.9`), on the same multi-core hardware server, dnstap source working in `multithreaded` mode can process `20K` dnstap messages per second, comparing to `13K` per second in `singlethreaded` mode (i.e. option `multithreaded` is disabled). However, if option `raw_data_only` was enabled, the `multithreaded` mode would be even slower than the `singlethreaded` mode (i.e. `50K` in `multithreaded` mode vs. `40K` in `singlethreaded` mode).

Reply:
Updated.